test/externalTests/foundry.py

```diff
@@ -0,0 +1,178 @@
+                        f.write(chunk)
+                run_cmd(f"chmod +x {foundryup_bin.resolve()}")
+                run_cmd(f"{foundryup_bin.resolve()}", self.env)
+            if not any(path == foundry_bin_dir for path in self.env["PATH"].split(os.pathsep)):
+                self.env["PATH"] += os.pathsep + str(foundry_bin_dir.resolve())
+        if self.setup_fn:
+            self.setup_fn(self.test_dir, self.env)
+
+    @staticmethod
+    def profile_name(preset: str):
+        """Returns foundry profile name"""
+        # Replace - or + by underscore to avoid invalid toml syntax
+        return re.sub(r'(\-|\+)+', '_', preset)
+
+    @TestRunner.on_local_test_dir
+    def clean(self):
+        """Clean the build artifacts and cache directories"""
+        run_cmd("forge clean")
+
+    @TestRunner.on_local_test_dir
+    def compiler_settings(self, solc_version: str, presets: Tuple[str] = AVAILABLE_PRESETS):
+        """Configure forge tests profiles"""
+
+        foundry_config_file = self.config.config_file
+        binary_type = self.config.solc.binary_type
+        binary_path = self.config.solc.binary_path
+        print(f"""Configuring Forge profiles...
+-------------------------------------
+Config file: {foundry_config_file}
+Binary type: {binary_type}
+Compiler path: {binary_path}
+-------------------------------------
+""")
+        # FIXME: Add support to solcjs. Currently only native solc is supported.
+        if binary_type == "solcjs":
+            raise RuntimeError(
+                "Solcjs binaries are currently not supported with Foundry. Please use `native` binary_type.")
+
+        profiles = []
+        for preset in presets:
+            name = self.profile_name(preset)
+            settings = settings_from_preset(preset, self.config.evm_version)
+            profiles.append(self.profile_tmpl.substitute(
+                name=name,
+                solc=binary_path,
+                evm_version=self.config.evm_version,
+                optimizer=str(settings["optimizer"]["enabled"]).lower(),
+                via_ir=str(settings["viaIR"]).lower(),
+                yul=str(settings["optimizer"]["details"]["yul"]).lower()
+            ))
+
+        with open(file=Path(self.test_dir) / foundry_config_file, mode="a", encoding="utf-8") as f:
+            for profile in profiles:
+                f.write(profile)
+
+        run_cmd("forge install", self.env)
+
+    @TestRunner.on_local_test_dir
+    def compile(self, solc_version: str, preset: str):
+        """Compile project"""
+
+        solc_short_version = ExternalTest.get_solc_short_version(solc_version)
+        settings = settings_from_preset(preset, self.config.evm_version)
+        print(f"""Using Forge profile...
+-------------------------------------
+Settings preset: {preset}
+Settings: {settings}
+EVM version: {self.config.evm_version}
+Compiler version: {solc_short_version}
+Compiler version (full): {solc_version}
+-------------------------------------
+""")
```

Comment:
You can use `dedent()` to keep this indented properly in the code.

Reply:
Nice! Thanks :)